roles/neutron-common/templates/usr/local/bin/neutron-reset

```diff
@@ -61,33 +62,69 @@ class Interface(object):
             return True
 
         # veth interfaces named tap... belong to Neutron
-        if re.match(r'^tap[0-9a-z]{8}-[0-9a-z]{2}$', self.ifname) and \
-                self.iftype() == 'veth':
-            return True
+        if re.match(r'^tap[0-9a-z]{8}-[0-9a-z]{2}$', self.ifname) and self.iftype() == 'veth':
+               return True
 
-        if re.match(r'^vxlan[\d]{1-4}$', self.ifname) and \
-                self.iftype() == 'vxlan':
+        if re.match(r'^vxlan\-[0-9]{1,4}$', self.ifname) and self.iftype() == 'vxlan':
             return True
 
-        if re.match(r'^brq[0-9a-z]{8}-[0-9a-z]{2}$', self.ifname) and \
-                self.iftype() == 'bridge':
+        if re.match(r'^brq[0-9a-z]{8}-[0-9a-z]{2}$', self.ifname) and self.iftype() == 'bridge':
             return True
 
         return False
 
+    @classmethod
+    def get_nova_interface(self, namespace=None):
```

Comment:
This method's name doesn't indicate it returns a list. Also, this appears to be a duplicate  of enumerate and is_nova_interface:
```nova_interfaces = [x for x in Interface.enumerate() if x.is_nova_interface()]```